### ISSUE TITLE
[BUG] Error when granting 'camera' permission for Chromium under MacOS

### ISSUE THREAD

**Issue by MrKotov** [microsoft/playwright#22644]
### System info
- Playwright Version: [v1.23]
- Operating System: macOS 13.3.1 22E261
- Browser: Chromium
- Other info: Using Playwright with TS and VSCode Extension for running tests

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

Currently I'm trying to test an application that invokes the user's camera.

I have attempted to grant context permissions through the `playwright.config.ts` file using `permissions: ['camera']` and chromium only configuration 
`playwright.config.ts`
```
import { defineConfig, devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
export default defineConfig({
  testDir: './tests',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    baseURL: 'https://10.1.2.52',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
    ignoreHTTPSErrors: true,
    permissions: ['camera']
  },

  /* Configure projects for major browsers */
  projects: [
   
    {
      name: 'chromium',
      use: { ...devices['iPhone 13 Pro']},
    },

    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: { ...devices['Pixel 5'] },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: { ...devices['iPhone 12'] },
    // },

    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: { ..devices['Desktop Chrome'], channel: 'chrome' },
    // },
  ],

  /* Run your local dev server before starting the tests */
  // webServer: {
  //   command: 'npm run start',
  //   url: 'http://127.0.0.1:3000',
  //   reuseExistingServer: !process.env.CI,
  // },
});
```
as well as adding the permissions via `beforeEach`  in the test: 
```
test.beforeEach(async ({ context }) => {
  await context.grantPermissions(['camera']);
});
```
Full test scenario:
`take-picture.spec.ts`
```
import { test, expect } from '@playwright/test';

test.beforeEach(async ({ context }) => {
  await context.grantPermissions(['camera']);
});

test('should take picture successfully', async ({ page, context }) => {
    await page.goto('https://10.1.2.52/');
    await page.getByRole('button', { name: 'Продължи' }).click();
    await expect(page).toHaveURL(/.*take-photo-front/);

  });
```
Currently what happens is that I'm getting the error that `Error: browserContext.newPage: Unknown permission: camera`. 
If I remove the config I get a similar window and can see the pop-up for the Permission grant but I don't seem to have a way of interacting with it.
![image](https://user-images.githubusercontent.com/16365036/234485433-cb1c5537-8376-42d1-b9f5-10028bb37dbb.png)

**Steps**
1. Run the test from VS Code via Playwright Plugin or via CLI

**Expected**

Camera permission should be granted to the Chromium Browser

**Actual**

Get test error `browserContext.newPage: Unknown permission: camera

**Comment by aslushnikov** [comment]
I can repro with the following HTML page:

```html

<script>
    document.addEventListener("DOMContentLoaded", () => {
      var but = document.getElementById("but");
      var video = document.getElementById("vid");
      var mediaDevices = navigator.mediaDevices;
      vid.muted = true;
      but.addEventListener("click", () => {
      
        // Accessing the user camera and video.
        mediaDevices
          .getUserMedia({
            video: true,
            audio: true,
          })
          .then((stream) => {
      
            // Changing the source of video to current stream.
            video.srcObject = stream;
            video.addEventListener("loadedmetadata", () => {
              video.play();
            });
          })
          .catch(alert);
      });
    });
</script>
<style>
    div {
      width: 500px;
      height: 400px;
      border: 2px solid black;
      position: relative;
    }
    video {
      width: 500px;
      height: 400px;
      object-fit: cover;
    }
</style>
<div>
    <video id="vid"></video>
</div>
<br />
<button id="but" autoplay>
    Open WebCam
</button>
```

And the following script:

```js
import { chromium, firefox } from '@playwright/test';

const browser = await chromium.launch({
  headless: false,
});

const page = await browser.newPage();
await page.context().grantPermissions(['camera']);
await page.goto('http://localhost:1234/webcam.html');
await page.locator('button').click();
```

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
